sdk/python/feast/client.py

```diff
@@ -1018,11 +1030,29 @@ def _infer_entity_rows(
         A list of EntityRow protos parsed from args.
     """
     entity_row_list = []
+    temp_dtype_storage = dict()
 
+    is_mixed_type = False
     for entity in entities:
         for key, value in entity.items():
+            if isinstance(value, list):
+                is_mixed_type = _is_mixed_type(value)
             # Infer the specific type for this row
             current_dtype = python_type_to_feast_value_type(name=key, value=value)
+
+            if is_mixed_type:
```

Comment:
I don't think this approach is optimal. I think we should just try and detect the first type and cast all remaining values. If there is a failure then there is a failure.

Reply:
hmm you mean letting this [function](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/type_map.py#L233) handle the failure?

Reply:
Yes

Reply:
Yup, that can handle it. If that's the case, we would not need `_is_mixed_type` function anymore. Will remove that then.